src/Compilers/CSharp/Portable/CodeGen/Optimizer.cs

```diff
@@ -1447,6 +1466,14 @@ public override BoundNode VisitSwitchDispatch(BoundSwitchDispatch node)
             return node.Update(boundExpression, node.Cases, node.DefaultLabel, node.LengthBasedStringSwitchDataOpt);
         }
 
+        public override BoundNode VisitLoweredIsPatternExpression(BoundLoweredIsPatternExpression node)
+        {
+            var statements = VisitSideEffects(node.Statements);
+            RecordBranch(node.WhenTrueLabel);
+            RecordBranch(node.WhenFalseLabel);
```

Comment:
I'm not sure what is the difference between RecordBranch and RecordLabel here and whether or not this is correct.

Reply:
After a closer look I realized in switch dispatch we're jumping _to_ those labels, but these are simply emitted as labels not branches, so I think these should use RecordLabel which I think is semantically the exact opposite of RecordBranch (switch PR already does this for arm labels).  Pushed a commit to test that out.